### ISSUE TITLE
error (rather than panic) on attempts to allocate absurdly large objects.

### ISSUE THREAD

**Issue by larceny-trac-import** [larcenists/larceny#343]
**Reported by: lth  on Sun Nov 21 06:00:00 1999**
143 v0.47 (991121 / lth)
Priority: low
Category: RTS / error handling

Currently, the system will signal an error appropriately if the
argument to make-vector, make-bytevector, or make-procedure is negative
or if it is a bignum but not if it is a nonnegative fixnum larger
than the max object size -- in that case, the system will panic.

The current behavior is inconsistent and unreasonably brutal.  It 
would be nice if some reasonable error could be signalled like in
the other cases.

**Comment by WillClinger** [comment]
It ought to be possible to convert this panic into something resembling a millicode exception (e.g. `mc_exception` in `src/Rts/IAssassin/millicode.c`).

**Comment by jkoser** [comment]
There seems to be a range of behaviors, at least on IAssassin:

```
Larceny v1.1a2 (Jun 14 2017 14:46:21, precise:Linux:unified)
larceny.heap, built on Wed Jun 14 14:51:11 EDT 2017

> (make-vector #x3fffff)
Larceny Panic: Can't allocate an object of size 16777216 bytes: max is 16777215 bytes.
```
```
> (make-vector #x023926ba)
Larceny Panic: Can't allocate an object of size 149199600 bytes: max is 16777215 bytes.
```
```
> (make-vector #x023926ba)
rlwrap: warning: larceny crashed, killed by SIGSEGV (core dumped).
```
```
> (make-vector #x1fffffff)
rlwrap: warning: larceny crashed, killed by SIGSEGV (core dumped).
```
```
> (make-vector #x20000000)
Error: make-vector-like: nonnaturalnumber: 536870912 
Entering debugger; type "?" for help.
debug>
```
The threshold between panic and segfault seems to be fuzzy, and I'm not sure why.  (Maybe ASLR?)  In any case, we should be able to guard against either behavior well before we call `gc_allocate`.

**Comment by WillClinger** [comment]
Fixed by changesets 72727d83c8c65a0ab42bd78a14cb8d5209545c9c and bb350e5f97a5901aeb424fd73a02f649518d3c52
